src/Augmint.ts

```diff
@@ -205,13 +213,109 @@ export class Augmint {
                 throw new Error("legacy contracts length mismatch!");
             }
         }
-        const options: ILoanManagerOptions = {
-            token: this.token, // FIXME: This should come from LoanManager contract's augmintToken property
-            rates: this.rates, // FIXME: This should come from LoanManager contract's rates property
-            ethereumConnection: this.ethereumConnection
-        };
         return legacyContracts.map(
-            (contract: DeployedContract<LoanManagerInstance>) => new LoanManager(contract.connect(this.web3), options)
+            (contract: DeployedContract<LoanManagerInstance>) =>
+                new LoanManager(contract.connect(this.web3), this.ethereumConnection)
+        );
+    }
+
+    public async getLoansForAccount(userAccount: string): Promise<Loan[]> {
+        return await Augmint.collectPromises(
+            this.getAllLoanManagers(),
+            loanManager => loanManager.getLoansForAccount(userAccount)
+        );
+    }
+
+    public async getLoanProducts(activeOnly: boolean): Promise<LoanProduct[]> {
+        return await Augmint.collectPromises(
+            this.getAllLoanManagers(),
+            loanManager => activeOnly ? loanManager.getActiveProducts() : loanManager.getAllProducts()
+        );
+    }
+
+    public async repayLoan(loan: Loan, repaymentAmount: Tokens, userAccount: string): Promise<Transaction> {
+        const loanManager: LoanManager = this.getLoanManagerByAddress(loan.loanManagerAddress)
+        const tokenAddress: string = await loanManager.tokenAddress;
+        const tokenContract: DeployedContract<TokenAEur> = this.deployedEnvironment.getContractFromAddress(
+            AugmintContracts.TokenAEur,
+            tokenAddress
+        );
+        const token: AugmintToken = new AugmintToken(tokenContract.connect(this.web3), this.ethereumConnection);
+        return loanManager.repayLoan(loan, repaymentAmount, userAccount, token);
+    }
+
+    public async newEthBackedLoan(loanProduct: LoanProduct, weiAmount: Wei, userAccount: string, minRate?:Tokens): Promise<Transaction> {
+        const loanManager: LoanManager = this.getLoanManagerByAddress(loanProduct.loanManagerAddress)
+        return loanManager.newEthBackedLoan(loanProduct, weiAmount, userAccount, minRate);
+    }
+
+    public async getAllLoans(): Promise<Loan[]> {
+        return await Augmint.collectPromises(
+            this.getAllLoanManagers(),
+            loanManager => loanManager.getAllLoans()
         );
     }
+
+    public async getLoansToCollect(): Promise<Loan[]> {
+        const allLoans: Loan[] = await this.getAllLoans();
+        return allLoans.filter((loan: Loan) => loan.isCollectable);
```

Comment:
you probably meant to actually invoke the `isCollectable()` function...

Reply:
isCollectable is not a function, it's an "accessor"